sysboundary/inflow.cpp

```diff
@@ -175,153 +107,155 @@ namespace SBC {
+      this->setCellDerivativesToZero(dperb, dmoments, stencil, component);
    }
-
-   void Inflow::fieldSolverBoundaryCondBVOLDerivatives(
-      FsGrid<std::array<Real, fsgrids::volfields::N_VOL>, FS_STENCIL_WIDTH>& volGrid, cint i, cint j, cint k,
-      cuint component) {
-      this->setCellBVOLDerivativesToZero(volGrid, i, j, k, component);
+   
+   void Inflow::fieldSolverBoundaryCondBVOLDerivatives(fsgrids::volspan vols,
+                                                       const fsgrid::FsStencil& stencil, cuint component) {
+      this->setCellBVOLDerivativesToZero(vols, stencil, component);
    }
-
+   
    void Inflow::vlasovBoundaryCondition(dccrg::Dccrg<SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid,
                                         const CellID& cellID, const uint popID, const bool doCalcMomentsV) {
       // This is a no-op because both template cell generation and block data copying takes place in
       // updateState() (at pre-set intervals only)
    }
-
+   
    void Inflow::setBFromTemplate(dccrg::Dccrg<SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid,
-                                 FsGrid<array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBGrid,
-                                 FsGrid<array<Real, fsgrids::bgbfield::N_BGB>, FS_STENCIL_WIDTH>& BgBGrid) {
+                                 fsgrids::perbspan perb,
+                                 fsgrids::bgbspan bgb,
+                                 fsgrids::technicalspan technical, FieldSolverGrid &fsgrid) {
       std::array<bool, 6> isThisCellOnAFace;
-
-      const std::array<FsGridTools::FsIndex_t, 3> gridDims(perBGrid.getLocalSize());
-      for (FsGridTools::FsIndex_t k = 0; k < gridDims[2]; k++) {
-         for (FsGridTools::FsIndex_t j = 0; j < gridDims[1]; j++) {
-            for (FsGridTools::FsIndex_t i = 0; i < gridDims[0]; i++) {
-               const auto coords = perBGrid.getPhysicalCoords(i, j, k);
-
+      const auto& gridSpacing = fsgrid.getGridSpacing();
+      const auto* localSize = &fsgrid.getLocalSize()[0];
+      for (fsgrid::FsIndex_t k = 0; k < localSize[2]; k++) {
+         for (fsgrid::FsIndex_t j = 0; j < localSize[1]; j++) {
+            for (fsgrid::FsIndex_t i = 0; i < localSize[0]; i++) {
+               const auto stencil = fsgrid.makeStencil(i, j, k);
+               const auto coords = fsgrid.getPhysicalCoords(i, j, k);
+   
                // TODO: This code up to determineFace() should be in a separate
                // function, it gets called in a lot of places.
                // Shift to the center of the fsgrid cell
                auto cellCenterCoords = coords;
-               cellCenterCoords[0] += 0.5 * perBGrid.DX;
-               cellCenterCoords[1] += 0.5 * perBGrid.DY;
-               cellCenterCoords[2] += 0.5 * perBGrid.DZ;
-
+               cellCenterCoords[0] += 0.5 * gridSpacing[0];
```

Comment:
You agree that this is a beautiful brainfart @markusbattarbee?

Reply:
Ah the block selection went somewhere. I meant down to the mpiGrid reflevel query and what we do with that just thereafter.

Reply:
No, this just extracts dx as a 2^-refLvl instead of using fsgrid's gridSpacing.

Reply:
Ah no ugh. We get an equivalent mpiGrid dx. duh.